openfe/protocols/openmm_rfe/_rfe_utils/relative.py

```diff
@@ -2346,21 +2347,99 @@ def _create_mdtraj_topology(self):
             at1_uniq = at1_hybrid_idx in self._atom_classes['unique_new_atoms']
             at2_uniq = at2_hybrid_idx in self._atom_classes['unique_new_atoms']
             if at1_uniq or at2_uniq:
-                if atom1.index in self._atom_classes['unique_new_atoms']:
-                    atom1_to_bond = added_atoms[atom1.index]
+                if at1_uniq:
```

Comment:
Bonds weren't being added properly here, which was causing issues with the roundtripping of these MDTraj topologies.

Turns out we didn't need to roundtrip, but we should really fix this whilst we're here.